shed/event_streams.py

```diff
@@ -1256,13 +1286,14 @@ def update(self, x, who=None):
 
 
 class QueryUnpacker(EventStream):
-    def __init__(self, db, child, fill=True):
+    def __init__(self, db, child, fill=True, **kwargs):
```

Comment:
This PR is becoming a catch all so I'll put it here. The basic idea is that we can't simultaneously query the database with provenance and unpack those queries (hand down documents and such).
Since we issue novel documents to track provenance and also to pass down data. If we track the provenance we won't have the queried documents, if we pass down the queried documents we have no idea how we got them.
Thus we break up the query into two nodes. The first queries for the data and returns the header UIDs (thus we know exactly what our query gave back). The second takes those uids, turns them into headers and then unpacks the headers into a stream of documents, giving us the data we needed.